Models/models.js

```diff
@@ -28,12 +28,22 @@ async function getCommentsModel(articleID) {
     return comments;
 }
 
-async function getAllArticlesModel() {
-    const results = await db.query(`
+async function getAllArticlesModel(queryObj) {
+    let sqlStr = `
     SELECT articles.article_id, articles.author, articles.topic, articles.created_at,
 articles.votes, articles.title, articles.article_img_url,
-COUNT(comments.article_id) as comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id GROUP BY articles.article_id ORDER BY articles.created_at DESC;
-    `);
+COUNT(comments.article_id) as comment_count FROM articles LEFT JOIN comments ON articles.article_id = comments.article_id
+    `;
+    if (queryObj.topic) {
+        const stringCheck = /^[a-z]+$/gi;
+        if (stringCheck.test(queryObj.topic)) {
+            sqlStr += ` WHERE articles.topic = '${queryObj.topic}'`
+        } else {
+            throw({code: "22P02"})
+        }
+    }
+    sqlStr += ` GROUP BY articles.article_id ORDER BY articles.created_at DESC;`
+    const results = await db.query(sqlStr);
     return results.rows;
 
 }
```

Comment:
Really nice use of queryString here. Rather than throwing a PSQL error ourselves here we should be making a greenlist for the valid topic queries and making sure that what we are passing in here is a valid topic. We could then return a Promise.reject if the query did not work. You can see an example of how to do this in the Variable Order By Statements section of the ncnotes on complex-queries.